### PR TITLE
72391 job scheduler part 2 fix

### PR DESCRIPTION
**Pull request by nilsgstrabo**

### CODE REVIEW

File: pkg/apis/deployment/radixcomponentname.go [hunk 1]
```diff
@@ -29,24 +29,52 @@ func (t RadixComponentName) ExistInDeploymentSpec(rd *v1.RadixDeployment) bool {
...
 	return false
 }
 
 // ExistInDeploymentSpecJobList checks if RadixDeployment has any job with this name
 func (t RadixComponentName) ExistInDeploymentSpecJobList(rd *v1.RadixDeployment) bool {
+	if job := t.findInDeploymentSpecJobList(rd); job != nil {
+		return true
+	}
+
+	return false
```
Comment (satr) [inline pkg/apis/deployment/radixcomponentname.go:59]: `return t.findInDeploymentSpecJobList(rd) != nil` - maybe?

File: pkg/apis/kube/kube.go [hunk 2]
```diff
@@ -36,6 +37,7 @@ const (
 	RadixJobTypeJob              = "job" // Outer job
 	RadixJobTypeBuild            = "build"
 	RadixJobTypeCloneConfig      = "clone-config"
+	RadixJobTypeJobSchedule      = "job-schedule"
```
Comment (satr) [inline pkg/apis/kube/kube.go:40]: Correct name? schedule or schedule**r** (I was not sure)
Used in `radix-job-scheduler`
image

File: pkg/apis/kube/roles.go [hunk 3]
```diff
@@ -150,6 +150,41 @@ func CreateManageSecretRole(appName, roleName string, secretNames []string, cust
 	return role
 }
 
+// GetRole creates a role for resource, that can manage a secret with predefined set of verbs
+func GetRole(appName, roleName, namespace string, apiGroups, resources, verbs []string, customLabels *map[string]string) *auth.Role {
```
Comment (satr) [inline pkg/apis/kube/roles.go:154]: Not used (created for rbac logic, which was removed)

**Review by satr — CHANGES_REQUESTED** [review]

File: pkg/apis/deployment/radixcomponentname.go (hunk 1 above)
Comment (nilsgstrabo) [reply pkg/apis/deployment/radixcomponentname.go:59]: Agree. Fixed

File: pkg/apis/kube/kube.go (hunk 2 above)
Comment (nilsgstrabo) [reply pkg/apis/kube/kube.go:40]: Perhaps "job-scheduler" is more correct. I have changed it.

File: pkg/apis/kube/roles.go (hunk 3 above)
Comment (nilsgstrabo) [reply pkg/apis/kube/roles.go:154]: Removed

File: pkg/apis/deployment/radixcomponentname.go (hunk 1 above)
Comment (satr) [reply pkg/apis/deployment/radixcomponentname.go]: I ment:
```
func (t RadixComponentName) ExistInDeploymentSpecJobList(rd *v1.RadixDeployment) bool {
	return t.findInDeploymentSpecJobList(rd) != nil 
}
```

Comment (nilsgstrabo) [reply pkg/apis/deployment/radixcomponentname.go]: Of course. Fixed

**Review by satr — APPROVED** [review]

**Pull request merged**